### PR TITLE
physics test level, conveyor fixes, implement some kind of drag

### PR DESCRIPTION
**Pull request by CMorooney**

### CODE REVIEW

File: src/assets.lua [hunk 1]
```diff
@@ -3,6 +3,7 @@ _G.assets = {
         tileset = "assets/images/tileset"
     },
     levels = {
-        test = "assets/levels/PlaydateProjectTest.ldtk"
+        test = "assets/levels/PlaydateProjectTest.ldtk",
+        physicsTest = "assets/levels/PlaydateProjectPhysicsTest.ldtk"
```
Comment (Franchovy) [inline src/assets.lua:7]: this is unused

File: src/assets/levels/PlaydateProjectTest.ldtk [hunk 2]
```diff
@@ -5198,6 +5198,372 @@
 				}
 			],
 			"__neighbours": [ { "levelIid": "9682b430-25d0-11ef-ba44-e352beaf71c7", "dir": "o" }, { "levelIid": "445f5ef0-25d0-11ef-ba44-7b5ec6701e46", "dir": "o" } ]
+		},
```
Comment (Franchovy) [inline src/assets/levels/PlaydateProjectTest.ldtk]: just for info: feel free to expand on this "physics test" test level to have multiple levels side-by-side to run consecutive tests. Think like a test lab :)

File: src/rooms/game.lua [hunk 3]
```diff
@@ -183,6 +184,8 @@ function Game:levelComplete(data)
     end
 
     local neighbors = LDtk.get_neighbours(currentLevelName, direction)
+   
+    if not neighbors then return end
```
Comment (Franchovy) [inline src/rooms/game.lua]: Not sure why you added this line. If there is no neighbour available, "neighbors" is an empty table (which evaluates to "true" in a boolean op).

Let's instead use: "assert(#neighbors > 0, "No neighbors available for: " .. currentLevelName .. " in direction:" .. direction)"

Comment (Franchovy) [reply src/rooms/game.lua]: Using an assert notifies us that there's an issue to address instead of ignoring it.

File: src/sprites/RigidBody.lua [hunk 4]
```diff
@@ -25,10 +24,13 @@ function RigidBody:init(entity, imageTable)
...
   self.velocity = gmt.vector2D.new(0, 0)
   self.inv_mass = 0.4
-  self.restitution = 0.4
+  self.restitution = 0
   self.static_friction = 0
   self.dynamic_friction = .12
+  self.air_friction = .14
   self.kinematic = false
+  self.maxFallSpeed = 13
+  self.maxXSpeed = 6
```
Comment (Franchovy) [inline src/sprites/RigidBody.lua]: `maxXSpeed` only seems to apply in conveyor belt logic. This naming might be misleading.

File: src/sprites/RigidBody.lua [hunk 5]
```diff
@@ -25,10 +24,13 @@ function RigidBody:init(entity, imageTable)
 
   self.velocity = gmt.vector2D.new(0, 0)
   self.inv_mass = 0.4
-  self.restitution = 0.4
+  self.restitution = 0
   self.static_friction = 0
   self.dynamic_friction = .12
+  self.air_friction = .14
   self.kinematic = false
+  self.maxFallSpeed = 13
```
Comment (Franchovy) [inline src/sprites/RigidBody.lua]: Tried removing `maxFallSpeed` altogether, since we have `air_friction`. Seems to work fine. Can we remove it?

**Review by Franchovy — CHANGES_REQUESTED** [review]

File: src/rooms/game.lua (hunk 3 above)
Comment (CMorooney) [reply src/rooms/game.lua]: the assert was throwing when I created a level with no neighbors for physics test which was not helpful, fwiw

File: src/sprites/RigidBody.lua (hunk 5 above)
Comment (CMorooney) [reply src/sprites/RigidBody.lua]: no -- `air_friction` applies to the X axis, this is a max amount for the Y to stop gravity from accumulating velocity to too high a Y value fro any given RB

**Comment by CMorooney** [comment]
@Franchovy made some updates to RB and elevator/parenting logic. not quite there but worth putting up to show the direction things are going.

I think I'm going to have to buckle down and make the Player a RB of some kind (probably kinematic) tomorrow to get everything playing nicely enough together....that and the parenting are a chunk of work

**Review by Franchovy — APPROVED** [review]

**Pull request merged**